### PR TITLE
Add `syntaxset` group FileType command after adding plugin

### LINKED ISSUE #523: lazy loading a plugin does not source the syntax files of plugin
- `nvim --version`: NVIM v0.6.0-dev+3de777d
- `git --version`: git version 2.32.0
- Operating system/version: several linux flavors
- Terminal name/version: windows terminal 1.9

### Steps to reproduce
i can encounter this issue with vim-svelte.
install the plugin, set it to lazy load, i am doign this using the ft = {'svelte'} property.
packercompile, and try opening a .svelte file, no syntax highlighting and no syntax file is sourced when checked using :scriptnames.
set the plugin to be loaded normally, and syntax highlighting works after a packercompile again.
### Actual behaviour
syntax files in opt directories arent sourced
### Expected behaviour
the syntax files should be sourced
### packer files
generic startup file from readme with 

<details>
<summary>Plugin specification file(s)</summary>

```lua 
return require'packer'.startup({function(use)
    use 'wbthomason/packer.nvim'
    use {
        "kyazdani42/nvim-web-devicons",
        module = "nvim-web-devicons",
    }
    use {
        'nvim-treesitter/nvim-treesitter',
        run = ":TSUpdate"
    }
    use {
        'SirVer/ultisnips',
        opt = true,
        requires = {'honza/vim-snippets', opt=true}
    }
    use {
        'kyazdani42/nvim-tree.lua',
        cmd = {"NvimTreeToggle", "NvimTree", "NvimTreeOpen", "NvimTreeClose"},
        requires = {'kyazdani42/nvim-web-devicons'}
    }
    use {
        'tpope/vim-commentary'
    }
    use {
        'nvim-lua/popup.nvim',
        module = "popup"
    }
    use {
        'nvim-lua/plenary.nvim',
        module = "plenary"
    }
    use {
        'windwp/nvim-autopairs',
    }
    use {
        'ayu-theme/ayu-vim',
        config = function()
            vim.cmd("colorscheme ayu")
        end
    }
    use {
        'hrsh7th/nvim-compe'
    }
    use {
        'itchyny/lightline.vim',
    }
    use {
        'nvim-telescope/telescope.nvim',
        cmd = "Telescope"
    }
    use {
        'neovim/nvim-lspconfig'
    }
    use {
        'tpope/vim-fugitive',
        cmd = {"G", "Fugitive"}
    }
    use {
        'fatih/vim-go',
        run = ":GoUpdateBinaries",
        ft = {"go", "gomod"}
    }
    use {
        'evanleck/vim-svelte',
        branch='main',
        ft = {'svelte', 'svx'},
        requires = {'pangloss/vim-javascript'}
    }
end,
config = {ensure_dependencies   = true}
})

```

</details>

<details>
<summary>packer log file</summary>
[INFO  Thu 17 Jun 2021 12:27:24 AM IST] .../site/pack/packer/start/packer.nvim/lua/packer/clean.lua:79: Already clean!
[DEBUG Thu 17 Jun 2021 12:27:28 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed kyazdani42/nvim-web-devicons
[DEBUG Thu 17 Jun 2021 12:27:28 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed ahmedkhalf/lsp-rooter.nvim
[DEBUG Thu 17 Jun 2021 12:27:28 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed windwp/nvim-ts-autotag
[DEBUG Thu 17 Jun 2021 12:27:29 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed terrortylor/nvim-comment
[DEBUG Thu 17 Jun 2021 12:27:29 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed folke/which-key.nvim
[DEBUG Thu 17 Jun 2021 12:27:29 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nacro90/numb.nvim
[DEBUG Thu 17 Jun 2021 12:27:29 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed numToStr/FTerm.nvim
[DEBUG Thu 17 Jun 2021 12:27:29 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed Pocco81/TrueZen.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed norcalli/nvim-colorizer.lua
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed simrat39/symbols-outline.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed kabouzeid/nvim-lspinstall
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed glepnir/galaxyline.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed neovim/nvim-lspconfig
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed hrsh7th/vim-vsnip
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-lua/popup.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed kevinhwang91/rnvimr
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed kevinhwang91/nvim-bqf
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed monaqa/dial.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed ChristianChiarulli/dashboard-nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed f-person/git-blame.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed glepnir/lspsaga.nvim
[DEBUG Thu 17 Jun 2021 12:27:30 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed phaazon/hop.nvim
[DEBUG Thu 17 Jun 2021 12:27:31 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed mfussenegger/nvim-dap
[DEBUG Thu 17 Jun 2021 12:27:31 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed lewis6991/gitsigns.nvim
[DEBUG Thu 17 Jun 2021 12:27:31 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed hrsh7th/nvim-compe
[DEBUG Thu 17 Jun 2021 12:27:33 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed folke/trouble.nvim
[DEBUG Thu 17 Jun 2021 12:27:33 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-telescope/telescope-fzy-native.nvim
[DEBUG Thu 17 Jun 2021 12:27:33 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed andymass/vim-matchup
[DEBUG Thu 17 Jun 2021 12:27:33 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-telescope/telescope.nvim
[DEBUG Thu 17 Jun 2021 12:27:38 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed windwp/nvim-autopairs
[DEBUG Thu 17 Jun 2021 12:27:39 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed christianchiarulli/nvcode-color-schemes.vim
[DEBUG Thu 17 Jun 2021 12:27:40 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-treesitter/nvim-treesitter
[DEBUG Thu 17 Jun 2021 12:27:42 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed metakirby5/codi.vim
[DEBUG Thu 17 Jun 2021 12:27:45 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed romgrk/barbar.nvim
[DEBUG Thu 17 Jun 2021 12:27:47 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed kyazdani42/nvim-tree.lua
[DEBUG Thu 17 Jun 2021 12:27:56 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed JoosepAlviste/nvim-ts-context-commentstring
[DEBUG Thu 17 Jun 2021 12:27:56 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-treesitter/playground
[DEBUG Thu 17 Jun 2021 12:27:56 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-telescope/telescope-project.nvim
[DEBUG Thu 17 Jun 2021 12:27:57 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed turbio/bracey.vim
[DEBUG Thu 17 Jun 2021 12:28:03 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed rafamadriz/friendly-snippets
[DEBUG Thu 17 Jun 2021 12:28:06 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed windwp/nvim-spectre
[DEBUG Thu 17 Jun 2021 12:28:13 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed nvim-lua/plenary.nvim
[DEBUG Thu 17 Jun 2021 12:28:19 AM IST] ...ite/pack/packer/start/packer.nvim/lua/packer/install.lua:23: Installed iamcco/markdown-preview.nvim
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/start/rnvimr/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/gitsigns.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/dashboard-nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/barbar.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/TrueZen.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/dial.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/bracey.vim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/hop.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/nvim-colorizer.lua/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/nvim-lspconfig/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/telescope.nvim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/nvim-dap/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/nvim-compe/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/vim-vsnip/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/start/nvim-treesitter/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/codi.vim/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/vim-matchup/doc
[INFO  Thu 17 Jun 2021 12:28:19 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/opt/nvim-tree.lua/doc
[INFO  Thu 17 Jun 2021 12:40:19 AM IST] ...e/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:316: All configured plugins are installed
[INFO  Thu 17 Jun 2021 12:41:11 AM IST] .../site/pack/packer/start/packer.nvim/lua/packer/clean.lua:79: Already clean!
[INFO  Thu 17 Jun 2021 12:41:15 AM IST] ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:84: Updating helptags for /home/rap/.local/share/nvim/site/pack/packer/start/packer.nvim/doc
[WARN  Mon 26 Jul 2021 03:40:46 PM IST 4169595286100] .../site/pack/packer/start/packer.nvim/lua/packer/clean.lua:79: Cleaning cancelled!
[WARN  Mon 26 Jul 2021 03:42:34 PM IST 4278173374300] ...m/site/pack/packer/start/packer.nvim/lua/packer/jobs.lua:87: Killing git due to timeout!
[WARN  Mon 26 Jul 2021 03:42:34 PM IST 4278186080400] ...m/site/pack/packer/start/packer.nvim/lua/packer/jobs.lua:87: Killing git due to timeout!
[WARN  Mon 26 Jul 2021 03:47:18 PM IST 4562018433600] ...ite/pack/packer/start/packer.nvim/lua/packer/display.lua:598: No plugin selected!
[WARN  Mon 26 Jul 2021 04:18:51 PM IST 6454840359500] ...ite/pack/packer/start/packer.nvim/lua/packer/display.lua:598: No plugin selected!
[WARN  Mon 26 Jul 2021 04:18:56 PM IST 6460132747800] ...ite/pack/packer/start/packer.nvim/lua/packer/display.lua:625: No plugin selected!
[WARN  Mon 26 Jul 2021 04:22:51 PM IST 6694681413500] ...ite/pack/packer/start/packer.nvim/lua/packer/display.lua:598: No plugin selected!
[WARN  Sat 31 Jul 2021 11:03:52 PM IST 1359083760700] ...e/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:742: You must run PackerCompile with profiling enabled first e.g. PackerCompile profile=true

</details>

<details>
<summary>packer compiled file</summary>
-- Automatically generated packer.nvim plugin loader code

if vim.api.nvim_call_function('has', {'nvim-0.5'}) ~= 1 then
  vim.api.nvim_command('echohl WarningMsg | echom "Invalid Neovim version for packer.nvim! | echohl None"')
  return
end

vim.api.nvim_command('packadd packer.nvim')

local no_errors, error_msg = pcall(function()

  local time
  local profile_info
  local should_profile = false
  if should_profile then
    local hrtime = vim.loop.hrtime
    profile_info = {}
    time = function(chunk, start)
      if start then
        profile_info[chunk] = hrtime()
      else
        profile_info[chunk] = (hrtime() - profile_info[chunk]) / 1e6
      end
    end
  else
    time = function(chunk, start) end
  end

local function save_profiles(threshold)
  local sorted_times = {}
  for chunk_name, time_taken in pairs(profile_info) do
    sorted_times[#sorted_times + 1] = {chunk_name, time_taken}
  end
  table.sort(sorted_times, function(a, b) return a[2] > b[2] end)
  local results = {}
  for i, elem in ipairs(sorted_times) do
    if not threshold or threshold and elem[2] > threshold then
      results[i] = elem[1] .. ' took ' .. elem[2] .. 'ms'
    end
  end

  _G._packer = _G._packer or {}
  _G._packer.profile_output = results
end

time([[Luarocks path setup]], true)
local package_path_str = "/home/rap/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?.lua;/home/rap/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/?/init.lua;/home/rap/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?.lua;/home/rap/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/?/init.lua"
local install_cpath_pattern = "/home/rap/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/lua/5.1/?.so"
if not string.find(package.path, package_path_str, 1, true) then
  package.path = package.path .. ';' .. package_path_str
end

if not string.find(package.cpath, install_cpath_pattern, 1, true) then
  package.cpath = package.cpath .. ';' .. install_cpath_pattern
end

time([[Luarocks path setup]], false)
time([[try_loadstring definition]], true)
local function try_loadstring(s, component, name)
  local success, result = pcall(loadstring(s))
  if not success then
    vim.schedule(function()
      vim.api.nvim_notify('packer.nvim: Error running ' .. component .. ' for ' .. name .. ': ' .. result, vim.log.levels.ERROR, {})
    end)
  end
  return result
end

time([[try_loadstring definition]], false)
time([[Defining packer_plugins]], true)
_G.packer_plugins = {
  ["ayu-vim"] = {
    config = { "\27LJ\2\n3\0\0\3\0\3\0\0056\0\0\0009\0\1\0'\2\2\0B\0\2\1K\0\1\0\20colorscheme ayu\bcmd\bvim\0" },
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/ayu-vim"
  },
  ["lightline.vim"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/lightline.vim"
  },
  ["nvim-autopairs"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/nvim-autopairs"
  },
  ["nvim-compe"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/nvim-compe"
  },
  ["nvim-lspconfig"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/nvim-lspconfig"
  },
  ["nvim-tree.lua"] = {
    commands = { "NvimTreeToggle", "NvimTree", "NvimTreeOpen", "NvimTreeClose" },
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/nvim-tree.lua"
  },
  ["nvim-treesitter"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/nvim-treesitter"
  },
  ["nvim-web-devicons"] = {
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/nvim-web-devicons"
  },
  ["packer.nvim"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/packer.nvim"
  },
  ["plenary.nvim"] = {
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/plenary.nvim"
  },
  ["popup.nvim"] = {
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/popup.nvim"
  },
  ["telescope.nvim"] = {
    commands = { "Telescope" },
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/telescope.nvim"
  },
  ultisnips = {
    after = { "vim-snippets" },
    after_files = { "/home/rap/.local/share/nvim/site/pack/packer/opt/ultisnips/after/plugin/UltiSnips_after.vim" },
    loaded = false,
    needs_bufread = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/ultisnips"
  },
  ["vim-commentary"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/vim-commentary"
  },
  ["vim-fugitive"] = {
    commands = { "G", "Fugitive" },
    loaded = false,
    needs_bufread = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/vim-fugitive"
  },
  ["vim-go"] = {
    loaded = false,
    needs_bufread = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/vim-go"
  },
  ["vim-javascript"] = {
    loaded = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/start/vim-javascript"
  },
  ["vim-snippets"] = {
    load_after = {
      ultisnips = true
    },
    loaded = false,
    needs_bufread = false,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/vim-snippets"
  },
  ["vim-svelte"] = {
    loaded = false,
    needs_bufread = true,
    path = "/home/rap/.local/share/nvim/site/pack/packer/opt/vim-svelte"
  }
}

time([[Defining packer_plugins]], false)
local module_lazy_loads = {
  ["^nvim%-web%-devicons"] = "nvim-web-devicons",
  ["^plenary"] = "plenary.nvim",
  ["^popup"] = "popup.nvim"
}
local lazy_load_called = {['packer.load'] = true}
local function lazy_load_module(module_name)
  local to_load = {}
  if lazy_load_called[module_name] then return nil end
  lazy_load_called[module_name] = true
  for module_pat, plugin_name in pairs(module_lazy_loads) do
    if not _G.packer_plugins[plugin_name].loaded and string.match(module_name, module_pat) then
      to_load[#to_load + 1] = plugin_name
    end
  end

  if #to_load > 0 then
    require('packer.load')(to_load, {module = module_name}, _G.packer_plugins)
    local loaded_mod = package.loaded[module_name]
    if loaded_mod then
      return function(modname) return loaded_mod end
    end
  end
end

if not vim.g.packer_custom_loader_enabled then
  table.insert(package.loaders, 1, lazy_load_module)
  vim.g.packer_custom_loader_enabled = true
end

-- Config for: ayu-vim
time([[Config for ayu-vim]], true)
try_loadstring("\27LJ\2\n3\0\0\3\0\3\0\0056\0\0\0009\0\1\0'\2\2\0B\0\2\1K\0\1\0\20colorscheme ayu\bcmd\bvim\0", "config", "ayu-vim")
time([[Config for ayu-vim]], false)

-- Command lazy-loads
time([[Defining lazy-load commands]], true)
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file NvimTreeToggle lua require("packer.load")({'nvim-tree.lua'}, { cmd = "NvimTreeToggle", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file NvimTree lua require("packer.load")({'nvim-tree.lua'}, { cmd = "NvimTree", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file G lua require("packer.load")({'vim-fugitive'}, { cmd = "G", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file Fugitive lua require("packer.load")({'vim-fugitive'}, { cmd = "Fugitive", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file NvimTreeOpen lua require("packer.load")({'nvim-tree.lua'}, { cmd = "NvimTreeOpen", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file NvimTreeClose lua require("packer.load")({'nvim-tree.lua'}, { cmd = "NvimTreeClose", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
pcall(vim.cmd, [[command -nargs=* -range -bang -complete=file Telescope lua require("packer.load")({'telescope.nvim'}, { cmd = "Telescope", l1 = <line1>, l2 = <line2>, bang = <q-bang>, args = <q-args> }, _G.packer_plugins)]])
time([[Defining lazy-load commands]], false)

vim.cmd [[augroup packer_load_aucmds]]
vim.cmd [[au!]]
  -- Filetype lazy-loads
time([[Defining lazy-load filetype autocommands]], true)
vim.cmd [[au FileType svelte ++once lua require("packer.load")({'vim-svelte'}, { ft = "svelte" }, _G.packer_plugins)]]
vim.cmd [[au FileType go ++once lua require("packer.load")({'vim-go'}, { ft = "go" }, _G.packer_plugins)]]
vim.cmd [[au FileType gomod ++once lua require("packer.load")({'vim-go'}, { ft = "gomod" }, _G.packer_plugins)]]
vim.cmd [[au FileType svx ++once lua require("packer.load")({'vim-svelte'}, { ft = "svx" }, _G.packer_plugins)]]
time([[Defining lazy-load filetype autocommands]], false)
vim.cmd("augroup END")
vim.cmd [[augroup filetypedetect]]
time([[Sourcing ftdetect script at: /home/rap/.local/share/nvim/site/pack/packer/opt/vim-go/ftdetect/gofiletype.vim]], true)
vim.cmd [[source /home/rap/.local/share/nvim/site/pack/packer/opt/vim-go/ftdetect/gofiletype.vim]]
time([[Sourcing ftdetect script at: /home/rap/.local/share/nvim/site/pack/packer/opt/vim-go/ftdetect/gofiletype.vim]], false)
time([[Sourcing ftdetect script at: /home/rap/.local/share/nvim/site/pack/packer/opt/vim-svelte/ftdetect/svelte.vim]], true)
vim.cmd [[source /home/rap/.local/share/nvim/site/pack/packer/opt/vim-svelte/ftdetect/svelte.vim]]
time([[Sourcing ftdetect script at: /home/rap/.local/share/nvim/site/pack/packer/opt/vim-svelte/ftdetect/svelte.vim]], false)
vim.cmd("augroup END")
if should_profile then save_profiles() end

end)

if not no_errors then
  vim.api.nvim_command('echohl ErrorMsg | echom "Error in packer_compiled: '..error_msg..'" | echom "Please check your config for correctness" | echohl None')
end
Post the contents of `packer_compiled.vim` here

</details>

### PR DESCRIPTION
**Pull request by deathlyfrantic**
Should fix #523.

### CODE REVIEW

**Comment by Iron-E** [comment]
Heh, I was about to post this. I am wondering if the other two `doautocmd` are necessary. My own testing seems to suggest that they aren't

File: lua/packer/load.lua [hunk 1]
```diff
@@ -103,6 +103,7 @@ packer_load = function(names, cause, plugins)
   elseif cause.event then
     cmd(fmt('doautocmd <nomodeline> %s', cause.event))
   elseif cause.ft then
+    cmd(fmt('doautocmd <nomodeline> FileType %s', cause.ft))
     cmd(fmt('doautocmd <nomodeline> %s FileType %s', 'filetypeplugin', cause.ft))
```
Comment (wbthomason) [inline lua/packer/load.lua:107]: As @Iron-E noted, if we're doing a group-less `FileType`  the other two `FileType` triggers become redundant. Would you mind removing them?

**Review by wbthomason — CHANGES_REQUESTED** [review]
This looks good - I think we only need to remove the redundant `FileType` triggers, and then this should be fine to merge.

File: lua/packer/load.lua (hunk 1 above)
Comment (deathlyfrantic) [reply lua/packer/load.lua]: Done!

**Review by wbthomason — APPROVED** [review]
Sorry for the delay in reviewing this; it looks good to me. Thanks again!

**Pull request merged**